keyvault/data-plane/azure-keyvault/src/main/java/com/microsoft/azure/keyvault/authentication/KeyVaultCredentials.java

```diff
@@ -119,10 +119,6 @@ public Response intercept(Chain chain) throws IOException {
 
         AuthenticationResult authResult = getAuthenticationCredentials(supportsPop, challengeMap);
 
-        if (authResult == null) {
-            return null;
-        }

```

Comment:
Sometimes this kind of change makes me nervous. It's possible the method night one day return null. What do you think?

Reply:
Hmm.. Fair enough. 
I was trusting the SpotBugs static analysis to surface those errors since it somehow knows when an object might be Null (ie. this bug description: https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#np-parameter-must-be-non-null-but-is-marked-as-nullable-np-parameter-must-be-nonnull-but-marked-as-nullable). I suppose it only applies to parameters though because of @Nullable/@NonNull attributes

I will revert this. :)